Add Trigger.dev V4 template

jingerpie opened the pull request:
Tried the existed v3 template, was not able to get it to work, so I made the new template base of the official guide:
https://trigger.dev/docs/self-hosting/docker#setup

There is problems remain unresolved:
Problem 1: Domain/URL not accessible when deploying via template

When I deploy directly using my template, the domain URL fails to open
However, when I manually copy the same docker-compose.yml and .env files into a new compose service and manually configure the domain, it works perfectly
This suggests the template deployment process isn't handling the domain configuration correctly

Problem 2: PostgreSQL username configuration error

When I try to change the PostgreSQL username (from the default), I get this error:

  Fatal: role "postgresql" does not exist

It appears the system is still looking for a role named "postgresql" instead of using my custom username

jingerpie committed:
feat: add trigger v4

jingerpie committed:
fix: add offcial env config

jingerpie committed:
feat: add auth env

jingerpie committed:
fix: pr error

Siumauricio commented:
I got error when importing in dokploy. 
<img width="1172" height="925" alt="image" src="https://github.com/user-attachments/assets/c3b0a9b6-d2ea-441d-b69e-c72eef751271" />

jingerpie committed:
fix: load error

jingerpie committed:
refactor: update docker-compose and template configuration

- Removed default values for environment variables in docker-compose.yml to enforce explicit configuration.
- Updated template.toml to streamline environment variable definitions and improve clarity.
- Adjusted service configurations for consistency across the stack, including logging and restart policies.
- Cleaned up commented sections in template.toml for better readability and maintainability.

jingerpie committed:
feat: add Trigger.dev V4 Web blueprint with Docker Compose setup

- Introduced a new blueprint for Trigger.dev V4 Web, including a comprehensive docker-compose.yml file for service orchestration.
- Added template.toml for environment variable configuration and service settings.
- Included SVG logo for branding.
- Updated meta.json to register the new blueprint with relevant metadata and links.

jingerpie committed:
feat: add Trigger.dev V4 Worker blueprint with Docker Compose setup

- Introduced a new blueprint for Trigger.dev V4 Worker, including a detailed docker-compose.yml for service orchestration.
- Added template.toml for environment variable configuration and service settings.
- Included SVG logo for branding.
- Updated meta.json to register the new blueprint with relevant metadata and links.

jingerpie committed:
feat: add DEPLOY_IMAGE_PLATFORM variable to Docker configuration

- Added DEPLOY_IMAGE_PLATFORM environment variable to docker-compose.yml for enhanced deployment flexibility.
- Set default value for DEPLOY_IMAGE_PLATFORM in template.toml to "linux/amd64" for consistency across environments.

jingerpie committed:
perf: simplify

jingerpie committed:
test: verify build & lock version

- Updated docker-compose.yml to use the latest Trigger.dev image version v4.0.4.
- Modified logging configuration values to be strings for consistency.
- Added registry_url variable to template.toml for improved service configuration.
- Defined a new registry service in template.toml with appropriate settings.

jingerpie committed:
chore: simplify worker build

- Set logging driver and options directly in docker-compose.yml for clarity.
- Updated image tags for supervisor and docker-proxy services to specific versions.
- Replaced TRIGGER_API_URL and OTEL_EXPORTER_OTLP_ENDPOINT with WEB_APP_URL in environment variables.
- Removed unnecessary environment variables from template.toml for simplification.

jingerpie committed:
chore: sort env

jingerpie committed:
chore: simplify web

- Replaced environment variables with direct values for database and ClickHouse configurations.
- Simplified template.toml by renaming variables and removing unnecessary entries for clarity.

jingerpie committed:
chore: sort env

jingerpie committed:
chore: update WEB_APP_URL in template.toml

- Changed WEB_APP_URL from "http://example.com:3000" to "http://example.com" for simplification.

jingerpie committed:
fix: registry missing env

jingerpie committed:
chore: remove unused TRIGGER_WORKER_TOKEN variable from template.toml

eonminion commented:
@bbreunig The trigger V4 has problem when you deploying it for first time, but after you deploy and letting it run for sometime, and redeploy without any config changes, it will work. 

"Removing all "custom" networks and letting dokploy create the network via "isolated deployment" made trigger.dev v4 run smoothly."
This is likely after you remove it, you deploy once again, making it to work. I am not sure if this is dokploy issue or the compose script issue. Can you help to validate this?

eonminion commented:
@Siumauricio Can you kindly review this once again since it hasnt been merge for 2 months. pLease let me know what else I should fix?

bbreunig commented:
@eonminion I redeployed it multiple times without it getting better. Only after removing the extra networks did it finally work. A "race condition" with dokploy networks is much more likely than what you described. Coolify has the same issue. So far, trigger worked flawlessly for the past 2 weeks without the other networks so I don't see a reason to keep them in the docker-compose.

eonminion commented:
@bbreunig Thanks for getting back to me. To clarify, is your solution to remove this entire section? One of the reasons I initially added docker-proxy was because it allows access to dokploy’s database deployment on the same server without ever needing to expose the database. In my project, I deploy the app and database within the same project and allow trigger dev to accept jobs to interact with the database using an internal address. 

networks:
~~docker-proxy:~~
~~name: docker-proxy~~
  supervisor:
    name: supervisor
  webapp:
    name: webapp

Siumauricio committed:
Update docker-compose.yml

Siumauricio committed:
Update docker-compose.yml

oRafael07 commented:
up

Siumauricio commented:
Close due to inactivity

LuizEdgar commented:
@jingerpie any chance to revive this?
